### ISSUE TITLE
Search against override rule w/ filter_curated_hits:true should ONLY return facet_counts for hits.

### ISSUE THREAD

**Issue by semanticdreamer** [typesense/typesense#1997]
## Description

A search w/ `filter_by` that triggers an override rule (e.g. by `override_tags`) BUT has `filter_curated_hits: true` in the response for `facet_counts` should ONLY have values for actual hits—NOT also from pinned (rule w/ `includes: []`) doc(s).

## Steps to reproduce

```shell
#!/bin/bash

TYPESENSE_BASE_URL="http://localhost:8108"
TYPESENSE_API_KEY=""

echo -e "\n\nDrop collection 'sfoti' if it exists…\n"
curl -k "${TYPESENSE_BASE_URL}/collections/sfoti" \
    -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" \
    -X DELETE

echo -e "\n\nCreate collection 'sfoti' with two properties, two of which are a facets…\n"
curl -k "${TYPESENSE_BASE_URL}/collections" -X POST \
    -H "Content-Type: application/json" \
    -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" -d '{
        "name": "sfoti",
        "fields": [
          {"name": "someprop", "index": true, "type": "string" },
          {"name": "somefacet", "index": true, "type": "string", "facet": true },
          {"name": "someotherfacet", "index": true, "type": "string", "facet": true }
        ]
    }'

echo -e "\n\nAdd documents to collection 'sfoti'…\n"
curl -k "${TYPESENSE_BASE_URL}/collections/sfoti/documents/import?action=upsert" -X POST \
    -H "Content-Type: text/plain" \
    -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" \
    -d '{"id": "4711", "someprop": "doc 4711", "somefacet": "sfa", "someotherfacet": "sofa"}
    {"id": "4712", "someprop": "doc 4712", "somefacet": "sfb", "someotherfacet": "sofb"}
    {"id": "4713", "someprop": "doc 4713", "somefacet": "sfc", "someotherfacet": "sofc"}'

echo -e "\n\nCreate override for collection 'sfoti' with tag rule to pin document 4712…\n"
curl -k "${TYPESENSE_BASE_URL}/collections/sfoti/overrides/pin-doc-4712" -X PUT \
    -H "Content-Type: application/json" \
    -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" -d '{
        "rule": {
            "tags": ["pin4712"]
        },
        "includes": [
            {"id": "4712", "position": 1}
        ],
        "filter_curated_hits": true
    }'

echo -e "\n\nSearch collection 'sfoti' with filter_by, facet_by, override_tags, filter_curated_hits…\n"
curl -k "${TYPESENSE_BASE_URL}/collections/sfoti/documents/search?limit=3&q=*&filter_by=somefacet:=sfa&facet_by=somefacet,someotherfacet&query_by=someprop&override_tags=pin4712" \
    -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" \
    | jq '.facet_counts[] | {field_name, values: [.counts[].value]}'
```

## Expected Behavior

Only facet counts for filtered doc.

```json
{
  "field_name": "somefacet",
  "values": [
    "sfa"
  ]
}
{
  "field_name": "someotherfacet",
  "values": [
    "sofa"
  ]
}
```

## Actual Behavior

Facet values from pinned doc.

```json
{
  "field_name": "somefacet",
  "values": [
    "sfb",
    "sfa"
  ]
}
{
  "field_name": "someotherfacet",
  "values": [
    "sofb",
    "sofa"
  ]
}
```

## Metadata

**Typesense Version**: v27.1

**OS**: cloud.typesense.org

**Comment by kishorenc** [comment]
Fixed in `28.0.rc15`.

**Comment by semanticdreamer** [comment]
I've successfully verified the fix in `28.0.rc15` on TS cloud. :white_check_mark: Thx a ton!
